### PR TITLE
Doc: Changelog for 0.5.1

### PR DESCRIPTION
**Pull request by LordAro**
Taken straight from the git log, and with extraneous commtis removed. Feel free to correct me.

### CODE REVIEW

File: docs/changelog.txt [hunk 1]
```diff
@@ -1,3 +1,19 @@
+0.5.1 (2020-05-10)
+------------------------------------------------------------------------
+- Add: Builtin functions roadtype() and tramtype()
+- Fix #116: towname used bits could be the same for sub-parts (#117)
```
Comment (glx22) [inline docs/changelog.txt:4]: I made a typo in the commit message, should be TownNames

File: docs/changelog.txt [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+- Fix #120: Value range of transported_last_month_pct should be 0..100 (#122)
+- Fix #121: check StorageOp register range only once
```
Comment (FLHerne) [inline docs/changelog.txt:6]: Maybe something more user-relevant? e.g.

Fix #121: STORE_PERM address limit should be 63

File: docs/changelog.txt [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+According to the NFO spec, there can be up to 63 each of road/tramtypes, rather than the 15 each that were allowed previously.
+OpenTTD also requires that the *combined* total of road and tram types be no more than 63, and that a label used for a roadtype cannot be used for a tramtype or vice versa.
+NML will not check for, nor enforce, these additional limitations.
+
+Other changes
+-------------
+- Add: Builtin functions roadtype() and tramtype()
+- Fix #116: TownNames used bits could be the same for sub-parts (#117)
+- Fix #120: Value range of transported_last_month_pct should be 0..100 (#122)
+- Fix #121: STORE_PERM address limit should be 63, not 15
```
Comment (frosch123) [inline docs/changelog.txt:16]: Uhm, what? The limit is 255. Is the message wrong, or is the code still wrong?

File: docs/changelog.txt [hunk 4]
```diff
@@ -1,3 +1,26 @@
+0.5.1 (2020-05-10)
+------------------------------------------------------------------------
+Among other, debian packaging related reasons, this release firms up the number of allowed Road/Tramtypes.
+
+Roadtypes
```
Comment (FLHerne) [inline docs/changelog.txt:5]: This release fixes a number of issues reported in 0.5.0.

N.B.: Although nmlc now permits up to 63 each of road and tramtypes, OpenTTD currently requires that the *combined* total not be more than 63 and that the same label not be used for both a roadtype and a tramtype. These limitations are not checked by nmlc.

File: docs/changelog.txt (hunk 3 above)
Comment (frosch123) [reply docs/changelog.txt:16]: Code is good, just changelog is wrong.

File: docs/changelog.txt [hunk 5]
```diff
@@ -1,3 +1,26 @@
...
+-------------
+- Add: Builtin functions roadtype() and tramtype()
+- Fix #116: TownNames used bits could be the same for sub-parts (#117)
+- Fix #120: Value range of transported_last_month_pct should be 0..100 (#122)
+- Fix #121: STORE_PERM address limit should be 63, not 15
+- Fix #134: Incorrect value of PALETTE_CC_RED
+- Fix #74: Update ottd_display_speed to match changes from OpenTTD r23945 (#75)
+- Fix: ANIM warnings in examples
+- Fix: Correct limit of available road/tramtypes
+- Fix: Produce, TownNames & TracktypeTable debug_print (#126, #127, #129)
```
Comment (FLHerne) [inline docs/changelog.txt:21]: Fix: Don't attempt to determine version from git when building from a release tarball.

File: docs/changelog.txt [hunk 6]
```diff
@@ -1,3 +1,24 @@
...
+These limitations are not checked by nmlc.
+
+- Add: Builtin functions roadtype() and tramtype()
+- Fix: Allow calculations for palette in spritelayout
+- Fix #116: TownNames used bits could be the same for sub-parts (#117)
+- Fix #120: Value range of transported_last_month_pct should be 0..100 (#122)
+- Fix #121: STORE_PERM address limit should be 255, not 15
+- Fix #134: Incorrect value of PALETTE_CC_RED
+- Fix #74: Update ottd_display_speed to match changes from OpenTTD r23945 (#75)
+- Fix: ANIM warnings in examples
```
Comment (FLHerne) [inline docs/changelog.txt:16]: Fix: Make the industry example do something, rather than crashing

**Pull request merged**